### PR TITLE
Refactor gum sample to ES6 code

### PR DESCRIPTION
**Pull request by ErikHellman**
**Description**
Refactor the gum sample to ES6.

**Purpose**
This is a small change to see that the tests still work and Travis accepts the changes. The following refactorings will be larger.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -35,7 +35,7 @@
     ],
     "max-len": [
       2,
-      80,
+      120,
```
Comment (henbos) [inline .eslintrc]: What's this? Line width?

File: src/content/getusermedia/gum/js/test.js [hunk 2]
```diff
@@ -5,98 +5,80 @@
...
+const webdriver = require('selenium-webdriver');
+const seleniumHelpers = require('webrtc-utilities').seleniumLib;
 
-var webdriver = require('selenium-webdriver');
-var seleniumHelpers = require('webrtc-utilities').seleniumLib;
+const path = '/src/content/getusermedia/gum/index.html';
+const url = (process.env.BASEURL ? process.env.BASEURL : ('file://' + process.cwd())) + path;
 
-test('Video width and video height are set on GUM sample', function(t) {
+test('Video width and video height are set on GUM sample', t => {
```
Comment (henbos) [inline src/content/getusermedia/gum/js/test.js:21]: Does test() or equivalent allow for "async t = { ... }" instead of "t => { ... }"?

File: src/content/getusermedia/gum/js/test.js [hunk 3]
```diff
@@ -5,98 +5,80 @@
...
 
-  driver.get((process.env.BASEURL ? process.env.BASEURL :
-      ('file://' + process.cwd())) +
-      '/src/content/getusermedia/gum/index.html')
-  .then(function() {
-    t.pass('Page loaded');
-  })
-  .then(function() {
-    return driver.wait(function() {
+  driver.get(url)
```
Comment (henbos) [inline src/content/getusermedia/gum/js/test.js:25]: ^--- ... if so, here and equivalent elsewhere:
await driver.get(url);

File: src/content/getusermedia/gum/js/test.js [hunk 4]
```diff
@@ -5,98 +5,80 @@
...
-      videoElement.getAttribute('videoHeight').then(function(h) {
-        height = h;
-        t.pass('Got videoHeight ' + h);
-        if (width && height) {
-          resolve([width, height]);
-        }
+    }, 30 * 1000))
+    // Check that there is a stream with a video track.
+    .then(() => driver.executeScript('return stream && stream.getTracks().length'))
+    .then(numberOfStreams => t.ok(numberOfStreams === 1, 'Stream exists and has one track'))
```
Comment (henbos) [inline src/content/getusermedia/gum/js/test.js:32]: Drive-by comment about code, unrelated to refactoring: There is no need to verify the number of tracks in a stream. If GUM() was not rejected and we have a stream, it will have a track. Let's not unittest that the browser is not buggy unless there are known issues (there aren't).

File: src/content/getusermedia/gum/js/main.js [hunk 5]
```diff
@@ -5,47 +5,44 @@
...
+  const errorElement = document.querySelector('#errorMsg');
+  errorElement.innerHTML += `<p>${msg}</p>`;
   if (typeof error !== 'undefined') {
     console.error(error);
   }
 }
 
-navigator.mediaDevices.getUserMedia(constraints).
-    then(handleSuccess).catch(handleError);
+navigator.mediaDevices
```
Comment (henbos) [inline src/content/getusermedia/gum/js/main.js:45]: Can we async/await and put the success/error code inline instead of use-once helper functions?

```
(async () => {
  try {
    const stream = await navigator.mediaDevices.getUserMedia(constraints);
    ... do whatever handleSuccess(stream) does ...
  } catch (e) {
    ... do whatever handleError(e); does ...
  }
})();
```

**Review by henbos — COMMENTED** [review]
Can we async/await? If so let me review again after changes, if there is a good reason not to or you want to do that separately I can approve the change as-is.

File: .eslintrc (hunk 1 above)
Comment (ErikHellman) [reply .eslintrc]: Yes. I changed this as it would otherwise make a lot of the code look really ugly with unnecessary line breaks.

I believe we should remove all the local rules here and use JavaScript standard or Google code-style directly, but that would be a different PR.

File: src/content/getusermedia/gum/js/test.js (hunk 2 above)
Comment (ErikHellman) [reply src/content/getusermedia/gum/js/test.js:21]: Yes. I plan to migrate to async/await in a later PR. Didn't want to make too much changes in this PR.

File: src/content/getusermedia/gum/js/test.js (hunk 4 above)
Comment (ErikHellman) [reply src/content/getusermedia/gum/js/test.js:32]: Yes, I have some opinions about the quality of the tests in general. Some of them are not relevant and other are testing the "wrong" things. Will fix in later PRs.

**Comment by ErikHellman** [comment]
The comments about async/await are relevant, but I'd like to make those changes later as it would mean much larger refactoring, as well as some heavy restructuring of the code. 

I plan to replace the custom selenium driver with something more up-to-date (also so we can show that testing WebRTC apps works with standard test runners).

File: src/content/getusermedia/gum/js/test.js (hunk 2 above)
Comment (henbos) [reply src/content/getusermedia/gum/js/test.js:21]: Ok sgtm!

File: src/content/getusermedia/gum/js/test.js (hunk 4 above)
Comment (henbos) [reply src/content/getusermedia/gum/js/test.js:32]: Cool

**Review by henbos — APPROVED** [review]
Ship it!
